### ISSUE TITLE
Should the help system consider synonyms?

### ISSUE THREAD

**Issue by ssiccha** [gap-system/gap#1727]
I get:
```
gap> DeclareSynonym("MyLength", Length );
gap> ?MyLength;
Help: no matching entry found
```

Wouldn't it be nice to have the help system show the results for `Length` instead? (The `??` command should stay unaffected IMO)

How else would you find the documentation of a function you only know the synonym of?

**Comment by alex-konovalov** [comment]
@ssiccha good point - I think if the synonym is documented, then it should. The place to add this would be `HELP_SEARCH_ALTERNATIVES` which already takes care about different spellings (e.g. Normaliser/Normalizer) and system setters and testers: 

https://github.com/gap-system/gap/blob/b8a6c4545ad37330f1fb9aaa2f6922c7c2b88ab4/lib/helpbase.gi#L139

**Comment by ssiccha** [comment]
Unfortunately `DeclareSynonym` is called with a string `name` and a GAP object `value` so it doesn't know the object `value`'s name.

Chris suggested to have a method `IDENTS_GVAR_VALUE` which returns a list of `[varname,value]` pairs, where `value` is unbound if the variable is unbound and to use that to find out a GVar's name.

I'll see whether I can get around to do that myself. :)

**Comment by alex-konovalov** [comment]
@ssiccha Alternatively, one could establish some global data structure which will be populated in calls to `DeclareSynonym` and then use it to return extra help search matches. This is how `TRANSATL` is used - see
https://github.com/gap-system/gap/blob/b8a6c4545ad37330f1fb9aaa2f6922c7c2b88ab4/lib/helpbase.gi#L95

**Comment by ssiccha** [comment]
Ah yes, `IDENTS_GVAR_VALUE` could then be used to populate the global data structure.
